Support proxy build arguments when building Docker images

Working behind a corporate HTTP proxy makes everything harder.
Here the docker-native goal fails on yum update && yum install
```
[INFO] Could not retrieve mirrorlist http://amazonlinux.default.amazonaws.com/2/core/latest/x86_64/mirror.list error was
12: Timeout on http://amazonlinux.default.amazonaws.com/2/core/latest/x86_64/mirror.list: (28, 'Failed to connect to amazonlinux.default.amazonaws.com port 80 after 5001 ms: Timeout was reached')

[ERROR] The command '/bin/sh -c yum update -y && yum install -y gcc gcc-c++ zlib-devel zip tar gzip && yum clean all' returned a non-zero code: 1

[ERROR] Failed to execute goal io.micronaut.build:micronaut-maven-plugin:3.5.3:docker-native (default-docker-native) on project laundry-micronaut-lambda: Execution default-docker-native of goal io.micronaut.build:micronaut-maven-plugin:3.5.3:docker-native failed: Could not build image: The command '/bin/sh -c yum update -y && yum install -y gcc gcc-c++ zlib-devel zip tar gzip && yum clean all' returned a non-zero code: 1
```
I managed to manually build the docker image by adding docker build args:
```
docker build --build-arg http_proxy=<proxy host>:<proxy port> --build-arg https_proxy=<proxy host>:<proxy port> .
```
Is it be possible to control the docker command arguments from Maven configuration?

--- a/micronaut-maven-plugin/src/site/asciidoc/examples/package.adoc
+++ b/micronaut-maven-plugin/src/site/asciidoc/examples/package.adoc
@@ -244,6 +244,24 @@ images. In this case, the `micronaut.runtime` property will also determine how t
 * Oracle Cloud Function: `mvn package -Dpackaging=docker-native -Dmicronaut.runtime=oracle_function`.
 * AWS Lambda (custom runtime): `mvn package -Dpackaging=docker-native -Dmicronaut.runtime=lambda`.
 
+For builds behind an HTTP or HTTPS proxy, `docker-native` automatically forwards the standard JVM proxy
+properties as Docker build arguments. The plugin maps `http.proxyHost`, `http.proxyPort`, `https.proxyHost`,
+`https.proxyPort`, and `http.nonProxyHosts` to the corresponding `HTTP_PROXY`, `HTTPS_PROXY`, and `NO_PROXY`
+build args, including their lowercase variants.
+
+[source,bash]
+----
+$ mvn package -Dpackaging=docker-native \
+    -Dhttp.proxyHost=proxy.example.com \
+    -Dhttp.proxyPort=8080 \
+    -Dhttps.proxyHost=proxy.example.com \
+    -Dhttps.proxyPort=8080 \
+    '-Dhttp.nonProxyHosts=localhost|127.0.0.1|*.example.com'
+----
+
+Prefer passing credential-bearing proxy values from the command line or CI secrets rather than committing them to
+the project POM.
+
 The image built can be customised using
 https://github.com/GoogleContainerTools/jib/tree/master/jib-maven-plugin#configuration[Jib]. In particular, you can set:
 

